src/main/java/agh/ics/oop/MapBoundary.java

```diff
@@ -0,0 +1,33 @@
+package agh.ics.oop;
+
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.Vector;
+
+//uznalem, ze dodanie mozliwosci rozroznienia na zwierzeta i trawe spowodowaloby spore zamieszanie - prawdopodobnie musialbym utworzyc nowy typ danych
+//np krotke, ktora skladalaby sie z Vector2d i typu obiektu na wspolrzednych, wymusilaby tez zmiane w positionChanged, a zatem i w interfejsie
+//poniewaz kepki trawy sie nie poruszaja, mozna w czasie liniowym, przy dodawaniu policzyc ich lowerLeft i upperRight
```

Comment:
lepiej - nie trzeba w ogóle tego liczyć w czasie liniowym, wystarczy na ślepo dodawać pozycje kolejnych traw do `MapBoundary`, a ono sobie będzie to porządkować w czasie logarytmicznym.